### PR TITLE
Add k8s v1.23 & v1.24 new metrics

### PR DESCRIPTION
**Pull request by kang-makes**
Skipping e2e because could fail as it is not in the entity definitions yet.

Metrics added to the table of the metrics we have to add to the spec files: #455

### CODE REVIEW

**Review by sigilioso — APPROVED** [review]
👍  LGTM

**Review by sigilioso — APPROVED** [review]
LGTM, I've just left a comment about versions checks in tests

File: src/controlplane/controlplane_test.go [hunk 1]
```diff
@@ -105,7 +105,11 @@ func Test_Scraper_Autodiscover_all_cp_components(t *testing.T) {
 			versionAsserter := asserter
 			// apiserverStorageObjects replaces etcObjectCounts in k8s versions above 1.23
 			if testutil.IsBelow(version, testutil.Testdata123) {
-				versionAsserter = versionAsserter.Excluding(exclude.Metrics("apiserverStorageObjects"))
+				versionAsserter = versionAsserter.Excluding(
+					exclude.Metrics("apiserverStorageObjects", "nodeCollectorEvictionsDelta"))
+			} else if testutil.IsBelow(version, testutil.Testdata124) {
+				versionAsserter = versionAsserter.Excluding(
+					exclude.Metrics("nodeCollectorEvictionsDelta", "etcdObjectCounts"))
```
[the remark this answers is not in the document]
Comment (marcsanmi) [reply src/controlplane/controlplane_test.go]: I thought `Excluding` was overwriting elements but first it copies them; much clearer this way, thanks! Updated dfad217ec4ea68d39fb945940efa2fa7dc404b95.

File: src/metric/definition.go [hunk 2]
```diff
@@ -189,6 +201,15 @@ var ControllerManagerSpecs = definition.SpecGroups{
...
 				Type:      sdkMetric.GAUGE,
 			},
+			{
+				Name: "nodeCollectorEvictionsDelta",
+				ValueFunc: prometheus.FromValueWithOverriddenName(
+					"node_collector_evictions_total",
+					"nodeCollectorEvictionsDelta",
+					prometheus.IgnoreLabelsFilter("zone"),
+				),
+				Type: sdkMetric.DELTA,
```
[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply src/metric/definition.go:211]: Usually I consider PDELTA to be safer and for example [here](https://github.com/newrelic/nri-kubernetes/blob/636ffad0f3c12095dfe6636267a61bb16af29808/src/metric/definition.go#L964) it is used that for the same reason

[the remark this answers is not in the document]
Comment (paologallinaharbur) [reply src/metric/definition.go:211]: further context -> https://github.com/newrelic/nri-nginx/pull/90

**Review by paologallinaharbur — APPROVED** [review]
everything looks correct!

**Pull request merged**